### PR TITLE
CCD-4371 Do not use TextValidator when validating a Document BaseType

### PR DESCRIPTION
**Pull request by marvincudjoe**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/CCD-4371

### Change description ###
Remove TextValidator on category_id
- Using TextValidator uses the regex pattern supplied for the document.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/ccd/domain/types/DocumentValidator.java [hunk 1]
```diff
@@ -149,10 +146,10 @@ private List<ValidationResult> validateCategoryId(final String dataFieldId,
             return Collections.emptyList();
         }
 
-        List<ValidationResult> validationResults =
```
Comment (padmapriyanalam) [inline src/main/java/uk/gov/hmcts/ccd/domain/types/DocumentValidator.java:152]: Removing the textValidator.validate() removes Max and Min validation along with Regex. Is this expected?

**Review by rajaram-nagarajan — APPROVED** [review]
LGTM

File: src/main/java/uk/gov/hmcts/ccd/domain/types/DocumentValidator.java (hunk 1 above)
Comment (marvincudjoe) [reply src/main/java/uk/gov/hmcts/ccd/domain/types/DocumentValidator.java:152]: Those checks work better for fields in the CaseFields tab. You can specify the min, max and regex for a field. For 
 Complex Types it's for for the complex type itself than the items inside it.

Comment (marvincudjoe) [reply src/main/java/uk/gov/hmcts/ccd/domain/types/DocumentValidator.java:152]: So for an item like category_id in a Document{} a Min, Max isn't needed. Unless it's made a requirement.

**Review by JamesCollettCGI — APPROVED** [review]

**Pull request merged**